nntrainer/tensor/cl_operations/blas_kernel_interface.cpp

```diff
@@ -131,15 +131,15 @@ void dotCl(Tensor const &input, Tensor const &m, Tensor &result, bool trans,
     }
     /// case2: (M * K) X (K * 1)
     else if (N == 1) {
-      trans ? sgemv_cl(data, mdata, rdata, dim2, dim1, lda)
-            : sgemv_cl(data, mdata, rdata, dim1, dim2, lda);
+      trans ? sgemv_cl(data, mdata, rdata, trans, dim2, dim1, lda)
+            : sgemv_cl(data, mdata, rdata, trans, dim1, dim2, lda);
```

Comment:
One suggestion! 
Since `trans` is now added as a parameter, handling `dim1` and `dim2` in the `sgemv_cl` function would be possible.

Reply:
```suggestion
      sgemv_cl(data, mdata, rdata, trans, dim2, dim1, lda)
```

and
```
// sgemv_cl
size_t dim1_size = transA ? sizeof(float) * dim2 : sizeof(float) * dim1;
size_t dim2_size = transA ? sizeof(float) * dim1 : sizeof(float) * dim2;
```

Reply:
The calls for `N == 1` and `M == 1` are a bit different. Also the work group size depends on the `dim1` and `dim2`. I think handling it before the function call would be better so that we can have a generalized function.